### DISCUSSION TITLE
How to have a job trigger to run when it's own execution ends

### DISCUSSION THREAD

**Discussion by securitypedant** [agronholm/apscheduler#725] [Q&A]
I am building a simple tool to enumerate folders, catalog images and present them in a Flash web UX.

I want a job that is started, runs, and then runs again once it's finished. So I want a trigger that simply restarts a job when an existing instance completes. Any ideas on how to configure this?

I was thinking I could set a trigger of 30 minutes and allow only one instance, then if it's taking over 30 minutes to enumerate all folders, the next job will not run and wait another 30 minutes. While this seems like an acceptable solution, I was wondering if there was a cleaner way to do it.

**Comment by agronholm** [comment]
```
def my_job():
    for _ in range(2):
        ...
```
